packages/flutter/lib/src/material/slider_theme.dart

```diff
@@ -1140,6 +1161,113 @@ class SliderThemeData with Diagnosticable {
     properties.add(
       DiagnosticsProperty<bool>('year2023', year2023, defaultValue: defaultData.year2023),
     );
+    properties.add(
+      DiagnosticsProperty<ValueIndicatorMultilineConfig?>(
+        'valueIndicatorMultilineConfig',
+        valueIndicatorMultilineConfig,
+        defaultValue: defaultData.valueIndicatorMultilineConfig,
+      ),
+    );
+  }
+}
+
+/// Configuration for value indicator multiline behavior.
+///
+/// Used with [SliderThemeData.valueIndicatorMultilineConfig] to control
+/// how multiline text is handled in slider value indicators.
+///
+/// See also:
+///
+///  * [SliderThemeData], which uses this to configure multiline behavior.
+///  * [Slider.label], which can contain newline characters for multiline text.
+@immutable
+class ValueIndicatorMultilineConfig with Diagnosticable {
+  /// Creates a configuration for multiline value indicators.
+  const ValueIndicatorMultilineConfig({this.enabled = true, this.maxLines, this.cornerPadding});
+
+  /// Whether multiline value indicators are enabled.
+  ///
+  /// When true, value indicators will dynamically size to accommodate
+  /// multiline text (text containing '\n' characters). When false,
+  /// all text will be treated as single-line for backward compatibility.
```

Comment:
Nit: Do you know of any cases where someone would actually want the original behavior? Maybe we could simplify things by not supporting the original behavior, unless someone opens an issue asking for it?

I'm assuming that the default behavior looks broken in most cases and so most people don't want it, but I could be wrong.